### DISCUSSION TITLE
Parsing freeform Text

### DISCUSSION THREAD

**Discussion by umgefahren** [alecthomas/participle#196] [Q&A]
I'm trying to build a parser with this library for [CookLang](https://github.com/cooklang/spec). How do I parse free form text and extract values from that?
Do I have to use the lexer? I'm quite new to all of this...

**Comment by alecthomas** [comment]
Participle's documentation is not really designed to explain the basics of parsing. However, if you follow the [tutorial](https://github.com/alecthomas/participle/blob/master/TUTORIAL.md), and look up some of the basic concepts such as the difference between lexing and parsing, you should be able to make it work.

Lexing basically splits the text into separate types of fragments (tokens), such as strings, numbers, whitespace, identifiers, etc. Parsing then enforces a structure on those tokens.
